src/renderer/components/delete-cluster-dialog/delete-cluster-dialog.tsx

```diff
@@ -0,0 +1,249 @@
+    this.removeContext();
+    this.changeCurrentContext();
+
+    try {
+      await saveKubeconfig(config, cluster.kubeConfigPath);
+      HotbarStore.getInstance().removeAllHotbarItems(cluster.id);
+      await requestMain(clusterDeleteHandler, cluster.id);
+    } catch(error) {
+      Notifications.error(`Cannot remove cluster, failed to process config file. ${error}`);
+      await requestMain(clusterClearDeletingHandler, cluster.id);
+    }
+
+    this.onClose();
+  }
+
+  isCurrentContext() {
+    return dialogState.config.currentContext == dialogState.cluster.contextName;
+  }
+
+  renderCurrentContextSwitch() {
+    if (!this.showContextSwitch) return null;
+    const { cluster, config } = dialogState;
+    const contexts = config.contexts.filter(context => context.name !== cluster.contextName);
+
+    const options = [
+      ...contexts.map(context => ({
+        label: context.name,
+        value: context.name,
+      })),
+    ];
+
+    return (
+      <div className="mt-4">
+        <Select
+          options={options}
+          onChange={({ value }) => this.newCurrentContext = value}
+          themeName="light"
+          className="ml-[1px] mr-[1px]"
+        />
+      </div>
+    );
+  }
+
+  getWarningMessage() {
+    const { cluster, config } = dialogState;
+    const contexts = config.contexts.filter(context => context.name !== cluster.contextName);
+
+    if (!contexts.length) {
+      return (
+        <p data-testid="no-more-contexts-warning">
+          This will remove the last context in kubeconfig. There will be no active context.
+        </p>
+      );
+    }
+
+    if (this.isCurrentContext()) {
+      return (
+        <p data-testid="current-context-warning">
+          This will remove active context in kubeconfig. Use drop down below to&nbsp;select a&nbsp;different one.
+        </p>
+      );
+    }
+
+    if (cluster.isInLocalKubeconfig()) {
+      return (
+        <p data-testid="internal-kubeconfig-warning">
+          Are you sure you want to delete it? It can be re-added through the copy/paste mechanism.
+        </p>
+      );
+    }
+
+    return (
+      <p data-testid="kubeconfig-change-warning">The contents of kubeconfig file will be changed!</p>
+    );
+  }
+
+  renderWarning() {
+    return (
+      <div className={styles.warning}>
+        <Icon material="warning_amber" className={styles.warningIcon}/>
+        {this.getWarningMessage()}
+      </div>
+    );
+  }
+
+  render() {
+    const { cluster, config, isOpen } = dialogState;
+
+    if (!cluster || !config) return null;
+
+    const contexts = config.contexts.filter(context => context.name !== cluster.contextName);
+
+    return (
+      <Dialog
+        className={styles.dialog}
+        isOpen={isOpen}
+        close={this.onClose}
+        onOpen={this.onOpen}
+      >
+        <div className={styles.dialogContent}>
+          <div>
+            Delete the <b>{cluster.getMeta().name}</b> context from <b>{cluster.kubeConfigPath}</b>?
```

Comment:
I still think that if the cluster is in a Lens local config we shouldn't display the kubeconfig path. If `cluster.isInLocalKubeconfig()` then:
```
            Delete the <b>{cluster.getMeta().name}</b> context from the Lens local config file?
```

Not a blocker though